### PR TITLE
Security Improvements

### PR DESCRIPTION
**Pull request by hypery2k**
see #220, still WIP

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "express": "4.16.2",
     "fresh": "0.5.2",
     "js-cookie": "2.1.4",
-    "morgan": "1.9.0",
+    "morgan": "^1.9.1",
```
Comment (Waschnick) [inline package.json]: Aren't fixed version better? Using ^ makes it uncertain which version will be used.

File: package.json [hunk 2]
```diff
@@ -120,10 +120,10 @@
     "karma-spec-reporter": "0.0.32",
     "karma-webpack": "3.0.0",
     "mkdirp": "0.5.1",
-    "mocha": "3.5.0",
+    "mocha": "^5.2.0",
```
Comment (Waschnick) [inline package.json]: mocha >3 has some problems with supertest, which we use to test the nodejs server. We would need to fix the node tests for this.

Comment (hypery2k) [reply package.json]: before npm version 5 this was right. The package-lock.json will enforce a reproducible and repeatable build ;)

File: package.json (hunk 1 above)
Comment (hypery2k) [reply package.json]: before npm version 5 this was right. The package-lock.json will enforce a reproducible and repeatable build ;)

**Comment by hypery2k** [comment]
thanks, but got it now ;)

**Pull request merged**